docs/src/main/paradox/jms.md

```diff
@@ -40,256 +40,236 @@ Gradle
+: @@snip (../../../../jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #run-map-source }
 
-### Using Topic with an JMS provider
 
-You can use JMS topic in a very similar way.
+### Receiving @extref[javax.jms.Message](javaee-api:javax.jms.Message)s from a JMS provider
 
-For the Sink :
+Create a @extref[javax.jms.Message](javaee-api:javax.jms.Message) source:
 
 Scala
-: @@snip ($alpakka$/jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala) { #create-topic-sink }
+: @@snip (../../../../jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala) { #create-jms-source }
 
 Java
-: @@snip ($alpakka$/jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #create-topic-sink }
+: @@snip (../../../../jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #create-jms-source }
 
-For the source :
+The `bufferSize` parameter controls the maximum number of messages to prefetch before applying backpressure.
+
+Run the source and specify the amount of messages to take:
 
 Scala
-: @@snip ($alpakka$/jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala) { #create-topic-source }
+: @@snip (../../../../jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala) { #run-jms-source }
 
 Java
-: @@snip ($alpakka$/jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #create-topic-source }
+: @@snip (../../../../jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #run-jms-source }
 
-Such sink and source can be started the same way as in the previous example.
+### Receiving @extref[javax.jms.Message](javaee-api:javax.jms.Message)s from a JMS provider with a selector
 
-### Running the example code
+Create a @extref[javax.jms.Message](javaee-api:javax.jms.Message) source specifying a [JMS selector expression](https://docs.oracle.com/cd/E19798-01/821-1841/bncer/index.html):
 
-The code in this guide is part of runnable tests of this project. You are welcome to edit the code and run it in sbt.
+Scala
+: @@snip (../../../../jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala) {             #create-jms-source-with-selector }
+
+Java
+: @@snip (../../../../jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #create-jms-source-with-selector }
+
+Verify that we are only receiving messages according to the selector:
 
 Scala
-:   ```
-    sbt
-    > jms/testOnly *.JmsConnectorsSpec
-    ```
+: @@snip (../../../../jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala) { #assert-only-odd-messages-received }
 
 Java
-:   ```
-    sbt
-    > jms/testOnly *.JmsConnectorsTest
-    ```
+: @@snip (../../../../jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #assert-only-odd-messages-received }
 
-### Using IBM MQ
```

Comment:
What happend to the IBM MQ section?

Reply:
Fixed. Maybe it was a little bit to late yesterday...